### PR TITLE
LA-48 Delete workgroup node in shared space

### PR DESCRIPTION
**Pull request by imGok**
demo_delete_file

### CODE REVIEW

File: data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart [hunk 1]
```diff
@@ -50,10 +50,7 @@ class SharedSpaceDocumentDataSourceImpl implements SharedSpaceDocumentDataSource
   );
 
   @override
-  Future<List<WorkGroupNode>> getAllChildNodes(
-      SharedSpaceId sharedSpaceId,
-      {WorkGroupNodeId parentNodeId}
-  ) {
+  Future<List<WorkGroupNode>> getAllChildNodes(SharedSpaceId sharedSpaceId, {WorkGroupNodeId parentNodeId}) {
```
Comment (hoangdat) [inline data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart:53]: please keep the current status

File: data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart [hunk 2]
```diff
@@ -81,10 +78,7 @@ class SharedSpaceDocumentDataSourceImpl implements SharedSpaceDocumentDataSource
   @override
   Future<List<WorkGroupNode>> copyToSharedSpace(CopyRequest copyRequest, SharedSpaceId destinationSharedSpaceId, {WorkGroupNodeId destinationParentNodeId}) {
     return Future.sync(() async {
-      return (await _linShareHttpClient.copyWorkGroupNodeToSharedSpaceDestination(
-            copyRequest.toCopyBodyRequest(),
-            destinationSharedSpaceId,
-            destinationParentNodeId: destinationParentNodeId))
+      return (await _linShareHttpClient.copyWorkGroupNodeToSharedSpaceDestination(copyRequest.toCopyBodyRequest(), destinationSharedSpaceId, destinationParentNodeId: destinationParentNodeId))
```
Comment (hoangdat) [inline data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart:81]: keep the current status

File: data/lib/src/network/linshare_http_client.dart [hunk 3]
```diff
@@ -189,4 +189,22 @@ class LinShareHttpClient {
...
+  Future<WorkGroupNodeDto> removeSharedSpaceNode(
+    SharedSpaceId sharedSpaceId,
+    WorkGroupNodeId sharedSpaceNodeId
+  ) async {
+    final workGroupNode = await _dioClient.delete(
+      Endpoint.sharedSpaces
+      .withPathParameter(sharedSpaceId.uuid)
+      .withPathParameter('nodes')
+      .withPathParameter(sharedSpaceNodeId.uuid)
+      .generateEndpointPath(),
```
Comment (hoangdat) [inline data/lib/src/network/linshare_http_client.dart:202]: indent

File: domain/lib/src/usecases/shared_space/remove_shared_space_node_interactor.dart [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+class RemoveSharedSpaceNodeInteractor {
+  final SharedSpaceDocumentRepository _sharedSpaceDocumentRepository;
+
+  RemoveSharedSpaceNodeInteractor(this._sharedSpaceDocumentRepository);
+
+  Future<Either<Failure, Success>> execute(List<WorkGroupNode> workGroupNodes) async {
+    try {
+      final workGroupNodesResult = {
+        for (var workGroupNode in workGroupNodes)
+          await _sharedSpaceDocumentRepository.removeSharedSpaceNode(workGroupNode.sharedSpaceId, workGroupNode.workGroupNodeId)
```
Comment (hoangdat) [inline domain/lib/src/usecases/shared_space/remove_shared_space_node_interactor.dart:45]: I don't want one delete executor error to impact others.

File: domain/test/usecases/shared_space/remove_shared_space_node_interactor_test.dart [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:testshared/fixture/shared_space_document_fixture.dart';
+
+import '../../mock/repository/mock_shared_space_document_repository.dart';
+
+void main() {
```
Comment (hoangdat) [inline domain/test/usecases/shared_space/remove_shared_space_node_interactor_test.dart:41]: Add test for the cases:
- delete 3 documents, with 1 failed, other still run -> return failed state
- delete 3 documents, all failed -> return failed state

File: domain/lib/src/usecases/shared_space/remove_shared_space_node_interactor.dart (hunk 4 above)
Comment (manhduc-nguyen) [reply domain/lib/src/usecases/shared_space/remove_shared_space_node_interactor.dart:45]: pls check `CopyMultipleFilesToSharedSpaceInteractor`, its very similar to this. you should use `Future.wait`

File: lib/presentation/localizations/app_localizations.dart [hunk 6]
```diff
@@ -408,42 +408,49 @@ class AppLocalizations {
   String get pick_the_destination {
     return Intl.message(
       'Pick the destination',
-      name: 'Pick the destination',
+      name: 'pick_the_destination',
```
Comment (manhduc-nguyen) [inline lib/presentation/localizations/app_localizations.dart:411]: sorry, I don't understand why you change this :(

File: lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_widget.dart [hunk 7]
```diff
@@ -325,4 +328,25 @@ class _WorkGroupNodesSurfingWidgetState extends State<WorkGroupNodesSurfingWidge
       ],
     );
   }
+
+  List<Widget> contextMenuFolderActionTiles(BuildContext context, WorkGroupFolder workGroupFolder) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_widget.dart:332]: pls make this function private

File: lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_widget.dart [hunk 8]
```diff
@@ -325,4 +328,25 @@ class _WorkGroupNodesSurfingWidgetState extends State<WorkGroupNodesSurfingWidge
...
+    return [];
+  }
+
+  List<Widget> contextMenuDocumentActionTiles(BuildContext context, WorkGroupDocument workGroupDocument) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_widget.dart:336]: same

File: lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_widget.dart [hunk 9]
```diff
@@ -325,4 +328,25 @@ class _WorkGroupNodesSurfingWidgetState extends State<WorkGroupNodesSurfingWidge
...
+    return [
+      removeDocumentAction(workGroupDocument),
+    ];
+  }
+
+  Widget removeDocumentAction(WorkGroupDocument workGroupDocument) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_widget.dart:342]: same

**Comment by imGok** [comment]
@hoangdat @manhduc-nguyen review pls

File: lib/presentation/view/modal_sheets/confirm_modal_sheet_builder.dart [hunk 10]
```diff
@@ -75,6 +75,7 @@ class ConfirmModalSheetBuilder {
 
   void show(context) {
     showModalBottomSheet(
+        useRootNavigator: true,
```
Comment (hoangdat) [inline lib/presentation/view/modal_sheets/confirm_modal_sheet_builder.dart:78]: Can you explain it?

**Review by hoangdat — APPROVED** [review]
LGTM!

**Review by manhduc-nguyen — APPROVED** [review]
LGTM

File: lib/presentation/view/modal_sheets/confirm_modal_sheet_builder.dart (hunk 10 above)
Comment (imGok) [reply lib/presentation/view/modal_sheets/confirm_modal_sheet_builder.dart:78]: image

It's to put the context menu widget in first layer

**Review by hoangdat — APPROVED** [review]

**Pull request merged**